code/player/TTTPlayer.Flashlight.cs

```diff
@@ -0,0 +1,172 @@
+using Sandbox;
+
+// TODO Fix flickering on close-range walls (with fast rotation)
+// TODO Add physics (avoid collision with walls or the playermodel)
+namespace TTTReborn.Player
+{
+    partial class TTTPlayer
+    {
+        private Flashlight _worldFlashlight;
+        private Flashlight _viewFlashlight;
+        private KeyframeEntity _flashlightHolder;
+
+        public bool HasFlashlightEntity
+        {
+            get
+            {
+                if (IsLocalPawn)
+                {
+                    return _viewFlashlight != null && _viewFlashlight.IsValid();
+                }
+
+                return _worldFlashlight != null && _worldFlashlight.IsValid();
+            }
+        }
+
+        public bool IsFlashlightOn
+        {
+            get
+            {
+                if (IsLocalPawn)
+                {
+                    return HasFlashlightEntity && _viewFlashlight.Enabled;
+                }
+
+                return HasFlashlightEntity && _worldFlashlight.Enabled;
+            }
+        }
+
+        public void ToggleFlashlight()
+        {
+            ShowFlashlight(!IsFlashlightOn);
+        }
+
+        public void ShowFlashlight(bool shouldShow, bool playSounds = true)
+        {
+            if (IsFlashlightOn == shouldShow)
+            {
+                return;
+            }
+
+            if (IsFlashlightOn)
+            {
+                if (IsServer)
+                {
+                    _worldFlashlight.TurnOff();
+                }
+                else
+                {
+                    _viewFlashlight.TurnOff();
+                }
+            }
+
+            if (IsServer)
+            {
+                ClientShowFlashlightLocal(To.Single(this), shouldShow);
+            }
+
+            if (shouldShow)
+            {
+                if (!HasFlashlightEntity)
+                {
+                    if (IsServer)
+                    {
+                        _worldFlashlight = new Flashlight();
+                        _worldFlashlight.EnableHideInFirstPerson = true;
+                        _worldFlashlight.Rotation = EyeRot;
+                        _worldFlashlight.Position = EyePos + EyeRot.Forward * 40f;
```

Comment:
Can we explain these magic numbers? i.e 40f? `const` them and name them based on what they are doing.